ADC/Core/Src/main.c

```diff
@@ -0,0 +1,150 @@
+/*
+ * 	Implement a routine that reads the internal voltage reference (ADC_IN17)
+ * 	and internal temperature sensor (ADC_IN18).
+ * 	The internal reference voltage must be in millivolts and the temperature in Celsius.
+ *
+ * 	1 - Polling		[/]
+ * 	2 - Interrupt	[ ]
+ * 	3 - DMA			[ ]
+ */
+
+#include "stm32l053xx.h"
+#include "macros.h"
+#include "init.h"
+
+uint8_t		ADC_error	= 0;	//error management flag
+uint16_t	measure 	= 0,	//AD read after a conversion
+			v_ref 		= 0;	//internal reference voltage measured (after calculation)
+int16_t		temperature	= 0;	//temperature in Celsius degrees measured (after calculation)
+
+void ADC_Config (uint8_t Channel);
+void Wait	 (uint16_t);
+
+int main (void)
+{
+	ADC_Init();
+
+	while(1)
+	{
+		ADC_Config(CH_INT_TEMP);			//internal temperature reading
+		ADC1->CR |= ADC_CR_ADSTART;
+		while(!(ADC1->ISR & ADC_ISR_EOC));
+		measure = ADC1->DR & ADC_DR_DATA;
+
+		//temperature calculation as in RM example
+		temperature = ((measure * VDD_APPLI / VDD_CALIB) - (int32_t) *TEMP30_CAL_ADDR);
+		temperature = temperature * (int32_t)(130 - 30);
+		temperature = temperature / (int32_t)(*TEMP130_CAL_ADDR - *TEMP30_CAL_ADDR);
+		temperature = temperature + 30;			//the temperature read is now stored in temperature
+
+
+		ADC_Config(CH_V_REF);				//internal reference voltage reading
+		ADC1->CR |= ADC_CR_ADSTART;
+		while(!(ADC1->ISR & ADC_ISR_EOC));
+		measure = ADC1->DR & ADC_DR_DATA;
+
+		//Vref calculation as in RM
+		v_ref = (3000 * (int32_t) (*VREFINT_CAL)) / measure;	//*3000 so we have Vref in mV
+
+		while(1);	//so we can hold the uC here after reading the temperature and Vref
+	}
+
+	return 0;
+}
+
+void ADC_Config (uint8_t Channel)
+{
+	ADC_error = 0;							//ADC management flag
```

Comment:
I got that you add this error flag to ensure that the user called the function with the right parameter, but there are other ways to do it. One of them is by using enumerators. Check ADC_Config function on TDR code to see how it's done.
You can also learn more on [this](https://www.learncpp.com/cpp-tutorial/45-enumerated-types/) link

Reply:
I liked it!